docs/labels.qmd

```diff
@@ -0,0 +1,403 @@
+
+-   **Ensuring minimal coupling to data object:** Because we are not embedding all metadata in the data frame, we avoid issues with large attributes. The only attribute attached by default will be the short variable label and the value labels. Everything else (notes, long descriptions, etc.) stays in the sidecar. This keeps the data frame S3 object clean and focused, which is important given that some R functions drop unknown attributes. We will document that the full metadata is always available via `recodeflow`’s sidecar and functions, even if not attached to the data frame.
+
+In summary, the label functionality is built with extensibility in mind: today it’s labels, tomorrow it could be definitions, provenance, etc. The design relies on the flexible sidecar metadata approach, avoiding heavy modifications to the core data object. This allows `recodeflow` to grow in metadata scope without large refactoring. We will maintain backward compatibility with projects that might not use new fields (extra columns can be optional), and ensure that any additional metadata fields do not break existing functions. By doing so, we keep `recodeflow` lean yet ready to capture rich data documentation as needed.
+
+## 8. Technical documentation and usability
+
+Clear documentation and sensible behavior are critical for user adoption of the new labeling features. We will provide comprehensive technical documentation and define how labels behave under various data transformations to ensure the feature is both machine-actionable and human-readable in practice.
+
+-   **Documentation of expected behavior:** The package vignettes and function references will describe how labels are preserved or modified during common operations. For example, if a user uses `dplyr::mutate()` to create a new variable from labeled variables, what happens? We will specify that by default, new variables will have no label (unless the user explicitly sets one), and existing labels will carry over to unmodified variables. If an existing labeled variable is transformed (e.g., scaled or recoded), the label may be dropped because the content has changed – unless our function `recode_with_table` handles it internally. We will integrate with known behaviors; for instance, when using haven::as_factor() on a `labelled` variable, the result is a factor with levels corresponding to the value labels, but the variable-level label is usually lost in that conversion. Users should be aware of this. In cases where R’s native or tidyverse behavior might strip labels, we will mention it and often provide a workaround or our own function. A section of the vignette could be “Working with labeled data: do’s and don’ts”, covering things like subsetting (which retains labels on the subset), combining data frames, pivoting, etc.
+
+-   **Preservation of metadata in transformations:** We will aim for labels to survive through data transformations as much as logically possible. For instance:
```

Comment:
Can you provide code examples of how you imaging this would work for each transformation type? Would we override the  dplyr's mutate method to preserve labels? Would we have a separate function that does this?

Reply:
I really don't know! 🤷 There is increasing awareness of this challenging issue within the data science communities. I was mostly flagging the issue, and we may want to leave this for future updates.